lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -390,7 +390,8 @@ export abstract class CacheManager implements ICacheManager {
                 return;
             }
 
-            if (!!homeAccountId && !this.matchHomeAccountId(entity, homeAccountId)) {
+            // homeAccountId can undefined, and we want to filter out cached items that have a homeAccountId of ""
```

Comment:
I would expand on this, with the details from @bgavrilMS comment above, e.g.:

```suggestion
            // homeAccountId can undefined, and we want to filter out cached items that have a homeAccountId of ""
            // because we don't want a client_credential request to return a cached token that has a home_account_id
```